crates/oxc_linter/src/rules/import/no_commonjs.rs

```diff
@@ -0,0 +1,349 @@
+    /// var mod = require("./mod");
+    /// ```
+    ///
+    /// but `module.exports` is reported as usual.
+    ///
+    /// ### Allow conditional require
+    ///
+    /// By default, conditional requires are allowed, If the `allowConditionalRequire` option is set to `false`, they will be reported.
+    ///
+    /// ### Allow primitive modules
+    ///
+    /// If `allowPrimitiveModules` option is set to true, the following is valid:
+    ///
+    /// ```js
+    /// module.exports = "foo";
+    /// module.exports = function rule(context) {
+    ///   return { /* ... */ };
+    /// };
+    /// ```
+    ///
+    /// but this is still reported:
+    ///
+    /// ```js
+    /// module.exports = { x: "y" };
+    /// exports.z = function bark() { /* ... */ };
+    /// ```
+    ///
+    NoCommonjs,
+    restriction
+);
+
+fn is_conditional(parent_node: &AstNode, ctx: &LintContext) -> bool {
+    let is_cond = matches!(
+        parent_node.kind(),
+        AstKind::IfStatement(_)
+            | AstKind::TryStatement(_)
+            | AstKind::LogicalExpression(_)
+            | AstKind::ConditionalExpression(_)
+    );
+    #[allow(clippy::redundant_else)]
+    if is_cond {
+        #[allow(clippy::needless_return)]
+        return true;
+    } else {
+        let Some(parent) = ctx.nodes().parent_node(parent_node.id()) else {
+            return false;
+        };
+        #[allow(clippy::needless_return)]
+        return is_conditional(parent, ctx);
+    }
+}
+/// <https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-commonjs.js>
+impl Rule for NoCommonjs {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+        Self {
+            allow_primitive_modules: obj
+                .and_then(|v| v.get("allowPrimitiveModules"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or(false),
+            allow_require: obj
+                .and_then(|v| v.get("allowRequire"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or(false),
+            allow_conditional_require: obj
+                .and_then(|v| v.get("allowConditionalRequire"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or(true),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::MemberExpression(member_expr) => {
+                // module.exports
+                let Some(property_name) = member_expr.static_property_name() else {
+                    return;
+                };
+
+                if !member_expr.object().is_specific_id("module")
+                    && !member_expr.object().is_specific_id("exports")
+                    && property_name != "exports"
+                {
+                    return;
+                }
+
+                if member_expr.object().is_specific_id("module") && property_name == "exports" {
```

Comment:
This if statement does redundant checks, see if you can refactor these two into a single `if` or `match`

Reply:
I remove previous check and that check is fit better as original rule.